skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/entities/user/FollowStateTracker.java

```diff
@@ -0,0 +1,98 @@
+
+    @Past
+    private Date creationDate;
+
+    @ManyToOne(cascade = CascadeType.REFRESH)
+    @JsonIgnoreProperties( { "followableSet", "followerSet"})
+//    @JsonManagedReference
+    private User followed;
+
+    @ManyToOne(cascade = CascadeType.REFRESH)
+    @JsonIgnoreProperties( { "followableSet", "followerSet"})
+//    @JsonManagedReference
+    private User follower;
+
+//    private LinkedHashMap<Object, Boolean> notifications = new LinkedHashMap<>();
+    private LinkedHashSet<Notification> notifications = new LinkedHashSet<>();
+
+    // Followable states
+    private FollowableStatus followedStatus;
+    private FollowableNotification followedNotifiable;
+
+    // Follower states
+    private FollowerStatus followerStatus;
+    private FollowerNotification followerNotifiable;
+
+    public FollowStateTracker(User followed, User follower) {
+        this.followed = followed;
+        this.follower = follower;
+        this.creationDate = new Date();
+        this.followedStatus = FollowableStatus.on;
+        this.followerStatus = FollowerStatus.on;
+        this.followedNotifiable = FollowableNotification.all;
+        this.followerNotifiable = FollowerNotification.all;
+    }
+
+    // User getters
+    public User getFollowed() { return this.followed; }
+    public User getFollower() { return this.follower; }
+
+    // User status Management
+    public FollowableStatus getFollowedStatus() { return this.followedStatus; }
+    public void setFollowedStatus(FollowableStatus followedStatus) { this.followedStatus = followedStatus; }
+
+    public FollowableNotification getFollowedNotifiable() { return this.followedNotifiable; }
+    public void setFollowedNotifiable(FollowableNotification followedNotifiable) { this.followedNotifiable = followedNotifiable; }
+
+    public FollowerStatus getFollowerStatus() { return this.followerStatus; }
+    public void setFollowerStatus(FollowerStatus followerStatus) { this.followerStatus = followerStatus; }
+
+    public FollowerNotification getFollowerNotifiable() { return this.followerNotifiable; }
+    public void setFollowerNotifiable(FollowerNotification followerNotifiable) { this.followerNotifiable = followerNotifiable; }
+
+    // Notification Management
+    public LinkedHashSet<Notification> getNotifications() { return this.notifications; }
+
+    public void pushNotifications(Set<Notification> notifications) {
+        notifications.forEach(notification -> {
+            notification.setRead(false);
+            this.notifications.add(notification);
+        });
+    }
+
+    public void popNotifications(Set<Notification> notifications) {
+        notifications.forEach( notification -> this.notifications.remove(notification) );
+    }
+
+    public void setNotificationStatus(Set<Notification> notifications, Boolean read) {
+        notifications.forEach(notification -> {
+            this.notifications.forEach(notif -> {
+                if (notification.getId() == notif.getId()) {
```

Comment:
In java 8 style, we would prefer something like:
```java
this.notifications.stream()
   .filter(notificationToUpdate -> notificationToUpdate.getId() == notification.getId())
   .forEach(notificationToUpdate -> notificationToUpdate.setRead(read));
```

Reply:
great ! implemented in next commit